### PR TITLE
refactor: sdk rhp

### PR DESCRIPTION
**Pull request by alexfreska**
- Updated sdk to latest core changes, updated structure.
- Follow up will migrate walletd methods into sdk, use sdk in walletd.

### CODE REVIEW

**Comment by alexfreska** [comment]
Current dependencies on/for this PR:
* **#519** <a href="https://app.graphite.dev/github/pr/SiaFoundation/web/519?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/SiaFoundation/web/519?utm_source=stack-comment).

**Comment by n8maninger** [comment]
I'm not sure I believe the `rhp` and `wallet` functions deserve their own package. It requires creating small utility packages (`api`, `utils`, `marshal`) to share code.

**Comment by alexfreska** [comment]
> I'm not sure I believe the `rhp` and `wallet` functions deserve their own package. It requires creating small utility packages (`api`, `utils`, `marshal`) to share javascript utilities.
> 
> If we do split them up into distinct packages, I don't think those packages should be "js-aware" and instead have concrete arguments and return values rather than `js.Value` and `api.Result`.

@n8maninger I moved everything into `main` 👍🏼

File: libs/sdk/src/utils/wasm_exec.js
Comment (n8maninger) [inline libs/sdk/src/utils/wasm_exec.js:1]: This should likely be copied from `$(go env GOROOT)/misc/wasm/wasm_exec.js` when the wasm file is built.

File: libs/sdk/src/utils/wasm_exec_tinygo.js
Comment (n8maninger) [inline libs/sdk/src/utils/wasm_exec_tinygo.js:1]: This should probably be deleted, no?

File: libs/sdk/src/utils/wasm_exec.js
Comment (n8maninger) [reply libs/sdk/src/utils/wasm_exec.js:1]: It's also interesting to note that there is a specific `wasm_exec_node.js` for use with node. I'm unsure if we've tested outside of the browser context, but that might be concerning.

**Review by ChrisSchinnerl — COMMENTED** [review]
I think after addressing Nate's comments this is good to go

File: libs/sdk/src/utils/wasm_exec.js
Comment (alexfreska) [reply libs/sdk/src/utils/wasm_exec.js:1]: https://github.com/SiaFoundation/web/pull/519/files#diff-32383f07904a9dacbd3550fd4d8935af3dfc2211ec49de70a1616dda04be0f1eR18

**Review by n8maninger — APPROVED** [review]

**Review by ChrisSchinnerl — APPROVED** [review]

**Pull request merged**